cmd/pulumi-test-language/tests.go

```diff
@@ -55,81 +60,134 @@ var languageTests = map[string]languageTest{
 
-			provider := snap.Resources[1]
-			assert.Equal(l, "pulumi:providers:simple", provider.Type.String(), "expected simple provider")
+					provider := snap.Resources[1]
+					assert.Equal(l, "pulumi:providers:simple", provider.Type.String(), "expected simple provider")
 
-			simple := snap.Resources[2]
-			assert.Equal(l, "simple:index:Resource", simple.Type.String(), "expected simple resource")
+					simple := snap.Resources[2]
+					assert.Equal(l, "simple:index:Resource", simple.Type.String(), "expected simple resource")
 
-			want := resource.NewPropertyMapFromMap(map[string]any{"value": true})
-			assert.Equal(l, want, simple.Inputs, "expected inputs to be {value: true}")
-			assert.Equal(l, simple.Inputs, simple.Outputs, "expected inputs and outputs to match")
+					want := resource.NewPropertyMapFromMap(map[string]any{"value": true})
+					assert.Equal(l, want, simple.Inputs, "expected inputs to be {value: true}")
+					assert.Equal(l, simple.Inputs, simple.Outputs, "expected inputs and outputs to match")
+				},
+			},
 		},
 	},
 	"l2-engine-update-options": {
 		providers: []plugin.Provider{&simpleProvider{}},
-		updateOptions: engine.UpdateOptions{
-			Targets: deploy.NewUrnTargets([]string{
-				"**target**",
-			}),
+		runs: []testRun{
+			{
+				updateOptions: engine.UpdateOptions{
+					Targets: deploy.NewUrnTargets([]string{
+						"**target**",
+					}),
+				},
+				assert: func(l *L, res result.Result, snap *deploy.Snapshot, changes display.ResourceChanges) {
+					requireStackResource(l, res, changes)
+					require.Len(l, snap.Resources, 3, "expected 2 resource in snapshot")
+
+					// Check that we have the target in the snapshot, but not the other resource.
+					stack := snap.Resources[0]
+					require.Equal(l, resource.RootStackType, stack.Type, "expected a stack resource")
+					provider := snap.Resources[1]
+					assert.Equal(l, "pulumi:providers:simple", provider.Type.String(), "expected simple provider")
+					target := snap.Resources[2]
+					require.Equal(l, "simple:index:Resource", target.Type.String(), "expected simple resource")
+					require.Equal(l, "target", target.URN.Name(), "expected target resource")
+				},
+			},
 		},
-		assert: func(l *L, res result.Result, snap *deploy.Snapshot, changes display.ResourceChanges) {
-			requireStackResource(l, res, changes)
-			require.Len(l, snap.Resources, 3, "expected 2 resource in snapshot")
-
-			// Check that we have the target in the snapshot, but not the other resource.
-			stack := snap.Resources[0]
-			require.Equal(l, resource.RootStackType, stack.Type, "expected a stack resource")
-			provider := snap.Resources[1]
-			assert.Equal(l, "pulumi:providers:simple", provider.Type.String(), "expected simple provider")
-			target := snap.Resources[2]
-			require.Equal(l, "simple:index:Resource", target.Type.String(), "expected simple resource")
-			require.Equal(l, "target", target.URN.Name(), "expected target resource")
+	},
+	"l2-multiple-tests": {
```

Comment:
l2-destroy might be a better name for this

Reply:
Agreed, that's much better :heavy_check_mark: